### PR TITLE
Implement Client::GetNotification().

### LINKED ISSUE #1034: Implement Client::GetNotification
This function should implement the [`Notifications: get`](https://cloud.google.com/storage/docs/json_api/v1/notifications/get) API.

This task is completed when:

- [ ] The new API has been added to Client.
- [ ] The new API has a completed doxygen comment block.
- [ ] There is a unit test for this API that checks the function works in a easy case.
- [ ] There is a unit test for this API that checks retries work correctly.
- [ ] There is a unit test for this API that checks errors are reported correctly.
- [ ] The unit tests run in the CI builds.
- [ ] There is an integration test for this API that runs against the emulator and against the GCS service.
- [ ] There is a code snippet showing how to use the API.
- [ ] The code snippet and the integration tests run in the CI builds.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #1034. It implements the API, the usual unit tests, extends
the integration test to call the API, adds a sample, and runs the sample
as part of the CI build.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1106)

### CODE REVIEW

File: google/cloud/storage/examples/storage_notification_samples.cc [hunk 1]
```diff
@@ -84,6 +84,24 @@ void CreateNotification(google::cloud::storage::Client client, int& argc,
...
+                     char* argv[]) {
+  if (argc != 3) {
+    throw Usage{"get-notification <bucket-name> <notification-id>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto notification_id = ConsumeArg(argc, argv);
+  //! [get notification] [START storage_print_pubsub_bucket_notification]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string notification_id) {
+    auto notification = client.GetNotification(bucket_name, notification_id);
```
Comment (frankyn) [inline google/cloud/storage/examples/storage_notification_samples.cc]: Use type instead of `auto`.

Comment (coryan) [reply google/cloud/storage/examples/storage_notification_samples.cc]: Fixed.

**Comment by coryan** [comment]
PTAL.

**Comment by coryan** [comment]
@houglum PTAL.

**Pull request merged**